### DISCUSSION TITLE
Attempt to build and register modifyed telegraf plugin

### DISCUSSION THREAD

**Discussion by strobs1621** [community/community#136672] [New to GitHub] [answered]
Hi, All!
I'm very new to GitHub, telegraf and programing in general. 
my goal is to modify existing telegraf plugin: ipmi_sensor. 
using multiple resources on web I completed these steps, hope I'm on correct track. 

1: Fork the Telegraf Repository
2: Clone Fork Locally
3. mkdir ipmi_sensor_1 in plugins/inputs/ and copied content of the folder ipmi_sensor to the folder ipmi_sensor_1
4. renamed ipmi_sensor.go to ipmi_sensor_1.go,    renamed ipmi_sensor_test.go to ipmi_sensor_test_1.go
5. replaced references to ipmi_sensor to ipmi_sensor_1 in  ipmi_sensor_1.go and  ipmi_sensor_test_1.go 
6. registered by creating ipmi_sensor_1.go in plugins/inputs/all
but make telegraf returned: 
plugins/inputs/all/ipmi_sensor_1.go:5:8: found packages ipmi_sensor_1 (connection.go) and ipmi_sensor (connection_test.go) in /etc/telegraf/telegraf/plugins/inputs/ipmi_sensor_1
make: *** [Makefile:138: build] Error 1

 Need some guidance here. 
Thank you

**Comment by ygor-sena** [comment, marked as answer]
Hi @strobs1621!

Hope that this helps: 

- Ensure that all `.go` files in your `ipmi_sensor_1` directory have the same package name, which should be `ipmi_sensor_1`;
- Specifically, open the `connection.go` and `connection_test.go` files in the `ipmi_sensor_1` directory and make sure the first line reads package `ipmi_sensor_1` in both files.

**Reply by strobs1621** [reply]
Thank you. It is compiled now.

**Comment by morteza1403** [comment]
"OK..
